client/my-sites/customer-home/locations/primary/index.jsx

```diff
@@ -2,43 +2,32 @@
  * External dependencies
  */
 import React from 'react';
-import { connect } from 'react-redux';
-import { isMobile } from '@automattic/viewport';
 
 /**
  * Internal dependencies
  */
 import GoMobile from 'my-sites/customer-home/cards/features/go-mobile';
 import ChecklistSiteSetup from 'my-sites/customer-home/cards/primary/checklist-site-setup';
 import QuickLinks from 'my-sites/customer-home/cards/primary/quick-links';
-import getSiteChecklist from 'state/selectors/get-site-checklist';
-import isEligibleForDotcomChecklist from 'state/selectors/is-eligible-for-dotcom-checklist';
-import isSiteChecklistComplete from 'state/selectors/is-site-checklist-complete';
-import { getSelectedSiteId } from 'state/ui/selectors';
 
-const Primary = ( { checklistMode, displayChecklist } ) => {
+const cardComponents = {
+	'feature.go-mobile-phones': GoMobile,
+	'primary.checklist-site-setup': ChecklistSiteSetup,
+	'primary.quick-links': QuickLinks,
+};
+
+const Primary = ( { checklistMode, cards } ) => {
 	return (
 		<div className="primary">
-			{ // "Go Mobile" is displayed on the primary location when viewed in smaller viewports, so folks
-			// can see it on their phone without needing to scroll.
-			isMobile() && <GoMobile /> }
-			{ displayChecklist && <ChecklistSiteSetup checklistMode={ checklistMode } /> }
-			{ ! displayChecklist && <QuickLinks /> }
+			{ cards &&
+				cards.map( ( card, index ) =>
+					React.createElement( cardComponents[ card ], {
+						key: index,
+						checklistMode: card === 'primary.checklist-site-setup' ? checklistMode : null,
```

Comment:
Unrelated to this PR, but I wish we could get rid of that `checklistMode`. Maybe we can refactor the `ChecklistSiteSetup` component in a follow-up so we don't need to inject it. That will simplify logic here and all locations would be the same (meaning we could have a single `Location` component).

Reply:
I'd maybe wait a bit on further refactors, @danhauk was exploring a number of variations/patterns that we might want to look into next. For example it could be that we move away from checklists all together.

Codewise I'm more of a fan of separate components, vs a really big smart component.

Reply:
Let's not bother with it for now then 👍